### ISSUE TITLE
String literals that are part of type definitions should not mutate

### ISSUE THREAD

**Issue by erangakm** [stryker-mutator/stryker#563]
For example: 

```
export type RequestGeneratorContext =
  Record<"requestGenerator", () => GraphQLRequest>;
```
``requestGenerator`` is attempted to be mutated and is being reported as untested. This shouldn't be mutated.

Tagging @mscharley

**Comment by mscharley** [comment]
I'm inclined to agree, we shouldn't mutate type definitions as a general rule - they can't be tested directly and structural typing will just mangle any attempts to test them indirectly.

**Comment by nicojs** [comment]
Hmm sounds convincing. @mscharley do you want to take this one?

**Comment by mscharley** [comment]
Yeah, I planned to. I'll probably get to it this weekend, shouldn't take too much to fix.

**Comment by simondel** [comment]
This has been released with `stryker-typescript@0.8.3`
